bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/FloureonThermostatHandler.java

```diff
@@ -0,0 +1,252 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.broadlinkthermostat.internal.handler;
+
+import static org.openhab.binding.broadlinkthermostat.internal.BroadlinkThermostatBindingConstants.*;
+
+import java.io.IOException;
+import java.time.LocalTime;
+import java.time.ZonedDateTime;
+
+import javax.measure.quantity.Temperature;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.library.types.*;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.mob41.blapi.FloureonDevice;
+import com.github.mob41.blapi.dev.hysen.AdvancedStatusInfo;
+import com.github.mob41.blapi.dev.hysen.BaseStatusInfo;
+import com.github.mob41.blapi.dev.hysen.SensorControl;
+import com.github.mob41.blapi.mac.Mac;
+import com.github.mob41.blapi.pkt.cmd.hysen.SetTimeCommand;
+
+/**
+ * The {@link FloureonThermostatHandler} is responsible for handling thermostats labeled as Floureon Thermostat.
+ *
+ * @author Florian Mueller - Initial contribution
+ */
+@NonNullByDefault
+public class FloureonThermostatHandler extends BroadlinkThermostatHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FloureonThermostatHandler.class);
+    private @Nullable FloureonDevice floureonDevice;
+
+    /**
+     * Creates a new instance of this class for the {@link FloureonThermostatHandler}.
+     *
+     * @param thing the thing that should be handled, not null
+     */
+    public FloureonThermostatHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * Initializes a new instance of a {@link FloureonThermostatHandler}.
+     */
+    @Override
+    public void initialize() {
+        super.initialize();
+        try {
+            blDevice = new FloureonDevice(host, new Mac(mac));
+            this.floureonDevice = (FloureonDevice) blDevice;
+        } catch (IOException e) {
+            logger.error("Could not find broadlinkthermostat device at Host {} with MAC {} ", host, mac, e);
+            updateStatus(ThingStatus.OFFLINE);
```

Comment:
Bindings should only log to error if something severe happened, like the detection of a bug in your code. See this link for a description of the log levels: https://www.openhab.org/docs/developer/guidelines.html#f-logging

You could remove the log message and use the message as a third argument to `updateStatus()` to display it in the UI.

Reply:
Moved message to updateStatus()